source/server/init_manager_impl.cc

```diff
@@ -38,7 +38,7 @@ void InitManagerImpl::initializeTarget(Init::Target& target) {
 
 void InitManagerImpl::registerTarget(Init::Target& target) {
   ASSERT(state_ != State::Initialized);
-  targets_.push_back(&target);
+  targets_.insert(&target);
```

Comment:
I think we should have an `ASSERT` that fires if we double register.

Reply:
That's my question in general. Isn't this a programming error? Why not just assert with the initial code and fix the calling code? Does this actually happen?

Reply:
@mattklein123 - I think this is happening somehow -- it's not clear to me exactly how -- in one of our deployments, through a series of events in which a listener is added, cluster warming finishes, and then the same listener is immediately re-added. I totally admit I don't understand the calling code nearly well enough, so it's definitely possible that the bug I'm fixing here isn't actually the bug we have!

@htuch - Adding the `ASSERT` definitely makes sense. Question about that: I'd kind of like to keep the test I've written, to make sure registering a target twice doesn't crash, but that relies on assertions being disabled (as in production) for the scope of the test. Is that possible?

Reply:
OK yeah I don't know without more debugging, but it seems like if you add the assert here, you are still going to violate it in your production deployment even though it won't crash? That seems wrong? So it seems like either it actually happens and there shouldn't be an assert, or there should only be an assert. :)

Reply:
Ok, per discussion here and in slack, I've limited this PR to just add the assertion and a couple comments.